python/lsst/meas/deblender/plugins.py

```diff
@@ -115,6 +116,205 @@ def __str__(self):
     def __repr__(self):
         return self.__str__()
 
+def _setPeakError(debResult, log, pk, cx, cy, filters, msg, flag):
+    """Update the peak in each band with an error
+
+    This function logs an error that occurs during deblending and sets the
+    relevant flag.
+    
+    Parameters
+    ----------
+    debResult: `lsst.meas.deblender.baseline.DeblenderResult`
+        Container for the final deblender results.
+    log: `log.Log`
+        LSST logger for logging purposes.
+    pk: int
+        Number of the peak that failed
+    cx: float
+        x coordinate of the peak
+    cy: float
+        y coordinate of the peak
+    filters: list of str
+        List of filter names for the exposures
+    msg: str
+        Message to display in log traceback
+    flag: str
+        Name of the flag to set
+
+    Returns
+    -------
+    None
+    """
+    log.trace("Peak {0} at ({1},{2}):{3}".format(pk, cx, cy, msg))
+    for fidx, f in enumerate(filters):
+        pkResult = debResult.deblendedParents[f].peaks[pk]
+        getattr(pkResult, flag)()
```

Comment:
This is using the API from the old deblender, which has a method `pkResult.setFlagname` for each flag and is used numerous times in the old deblender. I agree that this is not ideal, but has been in the codebase for years and is not worth fixing now since a lot of that code will be stripped out and the way we store and propagate result is likely to change significantly once we've officially adopted the new deblender.